src/run.rs

```diff
@@ -77,9 +84,26 @@ impl Runner {
         let source_dir = env::var_os("CARGO_MANIFEST_DIR")
             .map(PathBuf::from)
             .ok_or(Error::ProjectDir)?;
+        let source_manifest = dependencies::get_manifest(&source_dir);
 
         let features = features::find();
 
+        let path_dependencies = source_manifest
+            .dependencies
+            .iter()
+            .filter_map(|(name, dep)| {
+                if let Some(path) = dep.path.as_ref() {
+                    let path_dependency = PathDependency {
+                        name: name.to_owned(),
+                        normalized_path: path.canonicalize().ok()?,
```

Comment:
I cannot comment on the windows sidie of things here as I do no use windows myself and I do not have access to a windows machine to try that out. 
For linux the situation is as following:
Assuming a workspace with trybuild tests in `/home/users/diesel/diesel_compile_tests` and a path dependency to `diesel` via `../diesel/`, cargo will just resolve that path dependency as `/home/users/diesel/diesel_compile_tests/../diesel`, while rustc is reporting it as `/home/users/diesel/diesel/src/somefile.rs` in the error messages. That means we need to do some kind of file path normalization here, otherwise we cannot match both paths. 
If someone provides some samples of pathes coming from a similar error message on windows I can try to fix that.

Reply:
I think people prefer https://lib.rs/crates/dunce to do the canonicalization.